src/groovy/org/transmartproject/db/dataquery/clinical/variables/AbstractComposedVariable.groovy

```diff
@@ -27,5 +29,13 @@ import org.transmartproject.core.dataquery.clinical.ComposedVariable
 abstract class AbstractComposedVariable implements ComposedVariable {
     List<ClinicalVariable> innerClinicalVariables
 
+    String conceptPath
+
     abstract Object getVariableValue(DataRow<ClinicalVariableColumn, Object> clinicalVariableColumn)
+
+    @Override
+    ConceptKey getKey() {
+        def fullName = new ConceptFullName(conceptPath)
+        new ConceptKey(fullName.parts[0], fullName.toString())
+    }
```

Comment:
This looks too high in the hierarchy and it would be better to push it down, even if it means code duplication. The class is that we probably will want to support to across trial nodes with composed clinical variables (not possible right now: see `ClinicalVariableFactoryAcrossTrialsHelper`) and this class should then not have any knowledge of the logic for concept_dimension ontology terms.